addons/sale/controllers/portal.py

```diff
@@ -147,6 +148,18 @@ def portal_order_page(self, order_id, report_type=None, access_token=None, messa
         except (AccessError, MissingError):
             return request.redirect('/my')
 
+        # Force login if tokenization is required
+        if request.env['payment.acquirer'].sudo()._is_tokenization_required(
+            sale_order_id=order_sudo.id
+        ):
+            user_sudo = request.env.user
+            # The user needs to be logged if tokenization is required.
+            if user_sudo._is_public():
+                return request.redirect(
+                    # Escape special characters to avoid loosing original params when redirected
+                    f'/web/login?redirect={urllib.parse.quote(request.httprequest.full_path)}'
+                )
```

Comment:
So if I try to see a subscription SO, I've to be logged in, even if I don't want/have to pay ?
IMHO, this should at least be inside the `if order_sudo.has_to_be_paid()` :thinking: